### PR TITLE
Update samples to SDK version 2.3.0

### PR DESCRIPTION
**Pull request by torbsorb**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ There are two main categories of samples: **Camera** and **Applications**. The s
 ## Instructions
 
 1. [**Install Zivid Software**](https://zivid.atlassian.net/wiki/spaces/ZividKB/pages/59080712/Zivid+Software+Installation).
-Note: The samples require Zivid SDK v2 (minor version 2.2 or newer).
+Note: The samples require Zivid SDK v2 (minor version 2.3 or newer).
```
Comment (SatjaSivcev) [inline README.md:85]: This change is not needed.

File: source/CMakeLists.txt [hunk 2]
```diff
@@ -5,7 +5,7 @@ project(ZividCppSamples)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 include(CompilerOptions)
 
-set(ZIVID_VERSION 2.2.0)
+set(ZIVID_VERSION 2.3.0)
```
Comment (SatjaSivcev) [inline source/CMakeLists.txt:8]: Is this in conflict with our statement: Note: The samples require Zivid SDK v2 (minor version 2.2 or newer)?
I guess, we need to add stg like. To use version x.y.z change CMakeLists to x.y.z?

File: README.md (hunk 1 above)
Comment (torbsorb) [reply README.md:85]: True in theory, but it will fail on configuration because CMakeLists states 2.3.0

**Review by SatjaSivcev — CHANGES_REQUESTED** [review]
minor changes

File: README.md (hunk 1 above)
Comment (SatjaSivcev) [reply README.md:85]: Add a sentence to clarify then, see https://github.com/zivid/zivid-cpp-samples/pull/158#discussion_r615675488

Comment (torbsorb) [reply README.md:85]: I'm wrong. This seems to work as a "minimum" version match

File: source/CMakeLists.txt (hunk 2 above)
Comment (torbsorb) [inline source/CMakeLists.txt:8]: Based on the above we should not update this one

Comment (torbsorb) [reply source/CMakeLists.txt]: I checked and this check in CMake is not EXACT. Not entirely sure what it means (although it's explained here: https://cmake.org/cmake/help/latest/command/find_package.html#version-selection). In any case. I tested, and it works with both 2.2 and 2.3 if we leave it at 2.2. If we update to 2.3 then it only works with 2.3. It looks like it checks for minimum compatibility.

Comment (SatjaSivcev) [reply source/CMakeLists.txt]: That is great.

**Review by SatjaSivcev — APPROVED** [review]
LGTM

**Pull request merged**